app/interactors/add_movie_to_round.rb

```diff
@@ -3,26 +3,42 @@
 class AddMovieToRound
   include Hanami::Interactor
 
-  def initialize(import_service: ImportTmdbMovie.new)
-    @import_service = import_service
+  expose :movie, :round
+
+  def initialize(import_tmdb_movie_interactor: ImportTmdbMovie.new)
+    @import_tmdb_movie_interactor = import_tmdb_movie_interactor
   end
 
-  def call(round:, tmdb_id:)
+  def call(round:, tmdb_movie_id:)
     @round = round
-    @tmdb_id = tmdb_id
-
-    movie = Movie.find_by(tmdb_id: @tmdb_id)
-    movie = imported_movie if movie.blank?
+    @tmdb_movie_id = tmdb_movie_id
 
-    movie.errors.full_messages.each { |msg| error(msg) } unless @round.update(movie: movie)
+    fetch_movie
+    update_round
   end
 
   private
 
-  def imported_movie
-    result = @import_service.call(tmdb_movie_id: @tmdb_id)
-    fail! if result.failure?
+  def fetch_movie
+    found_movie = Movie.find_by(tmdb_id: @tmdb_movie_id)
+
+    @movie =
+      if found_movie.present?
+        found_movie
+      else
+        import_movie_result = import_tmdb_movie
+        fail! unless import_movie_result.successful?
```

Comment:
The change here to `unless` is because I think this is the only kind of thing where `unless` is clear and good. Following a pattern where it **always** goes in the "post-fix" position meaning "error"/"returning"//"breaking from loop", i.e. a "guard clause".